lib/perl/Genome/Qc/Command/BuildMetrics.pm

```diff
@@ -118,4 +144,138 @@ sub _calculate_haploid_coverage {
+                             pct_exc_overlap
+                             pct_exc_mapq
+                             pct_exc_baseq
+                         /;
+        for my $qc_metric_result(@$qc_metric_results) {
+            my $data = $self->_base_hash_ref_for_qc_metric_result($qc_metric_result);
+            $data->{'mean_target_coverage'} = $qc_metric_result->{MEAN_TARGET_COVERAGE};
+            $data->{'pct_usable_bases_on_target'} = $qc_metric_result->{PCT_USABLE_BASES_ON_TARGET};
+            $data->{'pct_target_bases_20x'} = $qc_metric_result->{PCT_TARGET_BASES_20X};
+            $data->{'pct_exc_off_target'} = $qc_metric_result->{PCT_EXC_OFF_TARGET};
+            $data->{'pct_exc_dupe'} = $qc_metric_result->{PCT_EXC_DUPE};
+            $data->{'pct_exc_overlap'} = $qc_metric_result->{PCT_EXC_OVERLAP};
+            $data->{'pct_exc_mapq'} = $qc_metric_result->{PCT_EXC_MAPQ};
+            $data->{'pct_exc_baseq'} = $qc_metric_result->{PCT_EXC_BASEQ};
+            $self->_add_microarray_vbid_result_metrics_to_hash_ref($qc_metric_result,$data);
+            push @data, $data;
+        }
+    } else {
+        die('Invalid TSV output QC profile!');
+    }
+    $self->_write_metrics_hash_refs_to_tsv_file(\@headers,\@data);
+    return 1;
+}
+
+sub _base_hash_ref_for_qc_metric_result {
+    my $self = shift;
+    my $qc_metric_result = shift;
+
+    my $build = Genome::Model::Build->get($qc_metric_result->{build_id});
+    my %data = (
+        sample_name => $build->model->subject->name,
+        build_id => $qc_metric_result->{build_id},
+        instrument_data_count => $qc_metric_result->{instrument_data_count},
+        instrument_data_ids => $qc_metric_result->{instrument_data_ids},
+        bam_path => $build->merged_alignment_result->bam_path,
+        pf_reads => $qc_metric_result->{PAIR}->{PF_READS},
+        aligned_bases => $qc_metric_result->{PAIR}->{PF_ALIGNED_BASES},
+        alignment_rate => ($qc_metric_result->{PAIR}->{PF_READS_ALIGNED} / $qc_metric_result->{PAIR}->{TOTAL_READS}),
+        duplication_rate => $qc_metric_result->{DUPLICATION_RATE},
+        first_of_pair_mismatch_rate => $qc_metric_result->{FIRST_OF_PAIR}->{PF_MISMATCH_RATE},
+        second_of_pair_mismatch_rate => $qc_metric_result->{SECOND_OF_PAIR}->{PF_MISMATCH_RATE},
+    );
+    return \%data;
+}
+
+#These results only exist from running SomaticValidation models on the same instrument data
+sub _add_microarray_vbid_result_metrics_to_hash_ref {
+    my $self = shift;
+    my $qc_metric_result = shift;
+    my $data = shift;
+
+    # Initialize as undefined
+    $data->{'microarray_chipmix'} = undef;
+    $data->{'microarray_freemix'} = undef;
+    
+    my $build = Genome::Model::Build->get($qc_metric_result->{build_id});
+
+    my %build_vbid_results;
+    for my $vbid (grep { $_->class eq 'Genome::InstrumentData::VerifyBamIdResult'} $build->results) {
+        # Not sure why but the results are not unique.  Duplicates are returned from build.
```

Comment:
There has been a recent change to `$build->results` so that only unique results are returned. See https://github.com/genome/genome/pull/1262